### ISSUE TITLE
testListenInvalidPort fails on OSX 

### ISSUE THREAD

**Issue by hutchig** [eclipse-vertx/vert.x#3444]
### Version

On the master branch but also on 4.0 milestones and 3.9.1

### Context

mvn clean install fails as the testcase testListenInvalidPort assumes that it is
not possible to bind to port 7 (echo). It is possible on OSX.

### Do you have a reproducer?

on OSX 'mvn clean install' on the master branch of this project 
commit 76418f20fbc41d4ad4e87455c90ba75d228248e7 (HEAD -> fix_testListenInvalidPort_On_OSX, my_fork/fix_testListenInvalidPort_On_OSX)

Will fail.

### Steps to reproduce

1. Get a macbook
2. Clone this repo
3. 'mvn clean install'

### Extra

mvn -v
Apache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-04T20:00:29+01:00)
Maven home: /Users/hutchig/apache-maven-3.6.1
Java version: 14.0.1, vendor: AdoptOpenJDK, runtime: /Library/Java/JavaVirtualMachines/adoptopenjdk-14.jdk/Contents/Home
Default locale: en_GB, platform encoding: UTF-8
OS name: "mac os x", version: "10.15.5", arch: "x86_64", family: "mac"

java -version
openjdk version "14.0.1" 2020-04-14
OpenJDK Runtime Environment AdoptOpenJDK (build 14.0.1+7)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 14.0.1+7, mixed mode, sharing)

**Comment by vietj** [comment]
@hutchig I'm using OSX and this never failed can you show the error you have ?

**Comment by hutchig** [comment]
What I saw was:
```
Starting test: Http1xTest#testListenInvalidPort 
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertFalse(Assert.java:64)
	at org.junit.Assert.assertFalse(Assert.java:74)
	at io.vertx.test.core.AsyncTestBase.assertFalse(AsyncTestBase.java:208)
	at io.vertx.test.core.AsyncTestBase.lambda$onFailure$0(AsyncTestBase.java:586)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$listen$5(HttpServerImpl.java:299)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:366)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:832)
Unhandled exception 
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertFalse(Assert.java:64)
	at org.junit.Assert.assertFalse(Assert.java:74)
	at io.vertx.test.core.AsyncTestBase.assertFalse(AsyncTestBase.java:208)
	at io.vertx.test.core.AsyncTestBase.lambda$onFailure$0(AsyncTestBase.java:586)
	at io.vertx.core.http.impl.HttpServerImpl.lambda$listen$5(HttpServerImpl.java:299)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:366)
	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:832)
```

**Comment by hutchig** [comment]
I have just spotted the extra "1x"! in the test name...investigating...

**Comment by hutchig** [comment]
[io.vertx.core.http.Http2Test.txt](https://github.com/eclipse-vertx/vert.x/files/4770953/io.vertx.core.http.Http2Test.txt)
[io.vertx.core.http.Http1xTest.txt](https://github.com/eclipse-vertx/vert.x/files/4770957/io.vertx.core.http.Http1xTest.txt)

**Comment by hutchig** [comment]
In my Apple OSX System Preferences: Firewall: Firewall Options I have got 
   [x] "Enable Stealthmode" 
set 
(I have never set this myself - so either it is Apple's or my Employers setting) 
this prevents 'ping' from working: I can verify with `ping localhost` not-working and working ). Checking now if this makes a difference to the test.

**Comment by hutchig** [comment]
Port 7 is not open on my machine according to sudo lsof.

`$ sudo lsof -nP -iTCP | grep LISTEN
UserEvent   116            root   76u  IPv6 0x8cb36b578a73eb27      0t0  TCP [fe80:4::aede:48ff:fe00:1122]:49153 (LISTEN)
UserEvent   116            root   77u  IPv6 0x8cb36b578a73f147      0t0  TCP [fe80:4::aede:48ff:fe00:1122]:49154 (LISTEN)
UserEvent   116            root   78u  IPv6 0x8cb36b578a73f767      0t0  TCP [fe80:4::aede:48ff:fe00:1122]:49155 (LISTEN)
UserEvent   116            root   79u  IPv6 0x8cb36b578a73fd87      0t0  TCP [fe80:4::aede:48ff:fe00:1122]:49156 (LISTEN)
UserEvent   116            root   80u  IPv6 0x8cb36b578a73d8c7      0t0  TCP [fe80:4::aede:48ff:fe00:1122]:49157 (LISTEN)
CrashPlan   118            root  228u  IPv6 0x8cb36b578e579d87      0t0  TCP 127.0.0.1:4244 (LISTEN)
vpnagentd   124            root    5u  IPv4 0x8cb36b578a754a37      0t0  TCP 127.0.0.1:29754 (LISTEN)
postgres    140        postgres    4u  IPv6 0x8cb36b578a73dee7      0t0  TCP *:5432 (LISTEN)
postgres    140        postgres    5u  IPv4 0x8cb36b578a755417      0t0  TCP *:5432 (LISTEN)
rapportd    583         hutchig    4u  IPv4 0x8cb36b579747f577      0t0  TCP *:64811 (LISTEN)
rapportd    583         hutchig    5u  IPv6 0x8cb36b5798c53d87      0t0  TCP *:64811 (LISTEN)
Spotify     823         hutchig   65u  IPv4 0x8cb36b57948dda37      0t0  TCP *:57621 (LISTEN)
Spotify     823         hutchig   67u  IPv4 0x8cb36b5790008417      0t0  TCP *:64891 (LISTEN)
eclipse   12546         hutchig  252u  IPv6 0x8cb36b57acb2b147      0t0  TCP 127.0.0.1:51757 (LISTEN)`

I wonder if, where the test is passing for @vietj if you get port 7 in the list?

**Comment by hutchig** [comment]
Confirmed fixed in master and 3.9
